### ISSUE TITLE
Allow web developer to relax restrictions?

### ISSUE THREAD

**Issue by noamr** [WICG/nav-speculation#143]
(This should by no means block existing efforts, more of a future thought)

Many of the behaviors restricted in a prerendering context are done so for preventing nonsensical behavior, or for preserving resources. This makes sense because many existing websites will become eligible for prerendering by default, without being aware and before they're able to make adjustments.

However, shouldn't we let websites who *are* aware of prerendering disable some of these restrictions, and manage things themselves using `document.prerendering` etc? For example, if a web-page requires service-worker registration, dedicated workers, `BroadcastChannel` and other restricted things and knows how to manage their effect on behavior while prerendering, do we have a reason to block them then? By allowing the restrictions to be relaxed by the web developer, we can (maybe?) make them a bit more strict by default and thus let developers enjoy the best of both worlds.

Suggesting that some meta-tag or [the SLM header](https://github.com/WICG/nav-speculation/issues/138#issuecomment-1035597813) should signal the UA to relax the non-mandatory restrictions.

**Comment by domenic** [comment]
Yeah, this makes sense. In general I think we're going to need to do a canvasing of all the restrictions to see what categories they fall into. E.g.:

- Necessary to avoid user annoyance
- Necessary for security
- Necessary for privacy
- Designed to reduce resource consumption
- Designed to reduce potentially-confusing global state modification
- Designed to reduce developers expecting a prerender to work, instead of treating it as an optimization (related to the last point)

This will also become important as we work to expand cross-origin. E.g. if some things are restricted for privacy reasons but not a security concern, then we could allow them on same-site cross-origin pages.

**Comment by noamr** [comment]
I believe also some of the restrictions can be relaxed based on the initiator of the prerendering (UA vs. document). For example, if an origin opted in to prerendering, they might as well be able to initiate service workers and post messages to broadcast channel.
